File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    
        var minPrice = 999999999;
        var maxProfit = 0;

        for (var i = 0; i < prices.length; i++){
            var p = prices[i];
            if (p < minPrice){
                minPrice = p;
            }
            
            var profit = p - minPrice;

            if (profit > maxProfit){
                maxProfit = profit;
            }
        }

        return maxProfit;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    
        int minPrice = 999999999;
        int maxProfit = 0;

        for (int i = 0; i < pricesSize; i++){
            int p = prices[i];
            if (p < minPrice){
                minPrice = p;
            }
            
            int profit = p - minPrice;

            if (profit > maxProfit){
                maxProfit = profit;
            }
        }

        return maxProfit;
}